editor/editor_vcs_api.h

```diff
@@ -0,0 +1,25 @@
+#ifndef EDITOR_VCS_API_H
+#define EDITOR_VCS_API_H
+
+#include "core/ustring.h"
+#include "core/object.h"
+
+class EditorVCSAPI {
+
+private:
+	String vcs_name;
+
+	static void register_to_editor(String vcs_name);
+
+	String _get_vcs_name();
```

Comment:
You do not need those private functions, the public ones are enough. The public function should be virtual though, as they will need to be overridden by extending classes.